cpp/include/nvtext/jaccard.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+namespace nvtext {
+/**
+ * @addtogroup nvtext_jaccard
+ * @{
+ * @file
+ */
+
+/**
+ * @brief Computes the jaccard similarity between individual rows
+ * in two strings columns
+ *
+ * The similarity is calculated between strings in corresponding rows
+ * such that `output[row] = J(input1[row],input2[row])`.
+ *
+ * The Jaccard index formula is https://en.wikipedia.org/wiki/Jaccard_index
+ * ```
+ *  J = |A ∩ B| / |A ∪ B|
+ *  where |A ∩ B| is number of common values between A and B
+ *  and |x| is the number of unique values in x.
+ * ```
+ *
+ * The computation here compares strings columns by treating each string as text (i.e. sentences,
+ * paragraphs, articles) instead of individual words or tokens to be compared directly. The
+ * algorithm applies a sliding window (size specified by the `width` parameter) to each string to
+ * form the set of tokens to compare within each row of the two input columns.
+ *
+ * These substrings are essentially character ngrams and used as part of the union and intersect
+ * calculations for that row. For efficiency, the substrings are hashed using the default
+ * MurmurHash32 to identify uniqueness within each row. Once the union and intersect sizes for the
+ * row are resolved, the Jaccard index is computed using the above formula and returned as a float32
+ * value.
+ *
+ * @code{.pseudo}
+ * input1 = ["the fuzzy dog", "little piggy", "funny bunny", "chatty parrot"]
+ * input2 = ["the fuzzy cat", "bitty piggy", "funny bunny", "silent partner"]
+ * r = jaccard_index(input1, input2)
+ * r is now [0.5, 0.15384616, 1.0, 0]
+ * @endcode
+ *
+ * If either input column's row is null, the output for that row will also be null.
+ *
+ * @throw std::invalid_argument if the `width < 5` or `input1.size() != input2.size()`
+ *
+ * @param input1 Strings column to compare with `input2`
+ * @param input2 Strings column to compare with `input1`
+ * @param width The character width used for apply substrings;
+ *              Default is 5 characters.
```

Comment:
I suppose this a Data Science question so I don't have a definitive answer. Anything less than this would be very impracticable since it would likely blow up memory and not yield good results. I picked this because that is the default used by the code I was given. I can certainly see using with greater than this and so wanted to make it a parameter. 
Do you think solving this should hold up this PR?

Reply:
I'd vote in favor of providing an unopinionated algorithm. n=2 seems just as natural as n=5, and the memory usage shouldn't be greatly affected. For a string of 100 characters, there are 96 sliding windows of size 5 and 99 sliding windows of size 2, so the number of windows doesn't grow much (input length - width + 1). And there are going to be more unique 5-character strings than 2-character strings after hash deduplication, so the postprocessing step should be even faster.

That said, this performance analysis could change a lot if we adopted a hashmap from cuCollections.

Reply:
To be clear, I'd vote for:
- no default width (user must provide it)
- require width >= 2 (not 5)

Reply:
Requesting input from @VibhuJawa and @ayushdg on the width parameter value range.

Reply:
1. I agree on `no default width `.

I have seen papers with `4`, `5` and `6` parameter ranges so far in papers . 

2. require width >= 2 , I think this seems reasonable.

Reply:
Ok, thanks. I've removed the default at set the minimum to 2.